src/devices/Common/Iot/Device/Multiplexing/IOutputSegment.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Device.Gpio;
+using System.Threading;
+
+namespace Iot.Device.Multiplexing
+{
+    /// <summary>
+    /// Interface that abstracts multiplexing over a segment of outputs.
+    /// </summary>
+    public interface IOutputSegment : IDisposable
+    {
+        /// <summary>
+        /// Length of segment (number of outputs)
+        /// </summary>
+        int Length { get; }
+
+        /// <summary>
+        /// Segment values.
+        /// </summary>
+        PinValue this[int index] { get; }
+
+        /// <summary>
+        /// CancellationToken for segment.
+        /// </summary>
+        CancellationToken CancellationToken { get; }
+
+        /// <summary>
+        /// Writes a PinValue to a multiplexed output.
+        /// Does not perform a latch.
+        /// </summary>
+        void Write(int output, PinValue value);
+
+        /// <summary>
+        /// Writes an int (processed as one of more bytes) to a multiplexed output.
+        /// Written one byte at a time, left to right. Least significant bit will written to index 0.
+        /// Does not perform a latch.
+        /// </summary>
+        void Write(int value);
+
+        /// <summary>
+        /// Turns off all outputs.
+        /// Performs a latch.
+        /// </summary>
+        void Clear();
+
+        /// <summary>
+        /// Displays segment until token receives a cancellation signal, possibly due to a specificated duration.
+        /// As appropriate for a given implementation, performs a latch.
+        /// </summary>
+        void Display(TimeSpan time);
```

Comment:
Same here, perhaps have an API that is a bit more descriptive with just the name? I guess that even by reading the summary here I'm not sure if the idea here is that not until you call Display then all previous Writes haven't really been applied?

Reply:
Right. Would `Update` be a better term than `Write`?